ci/benchmark_ai.sh

```diff
@@ -0,0 +1,218 @@
+ggregate_report_error aggregate_report_rate aggregate_report_bandwidth aggregate_report_stddev < <(tail -n1 /mxnet-model-server/load-test/report/$report)
+
+    echo -e "{\
+        Throughput_concurrency_${workers}_req_${requests} :$aggregate_report_rate,\
+        Average_latency_concurrency_${workers}_req_${requests} :$average,\
+        Median_latency_concurrency_${workers}_req_${requests} :$aggregate_report_median,\
+        P90_latency_concurrency_${workers}_req_${requests} :$aggregate_report_90_line,\
+        Error_rate_concurrency_${workers}_req_${requests} :$aggregate_report_error}\n"
+        IFS=$OLDIFS
+}
+
+#start the container test
+start_mms_cpu(){
+    docker run  --name mms -v /mxnet-model-server:/mxnet-model-server  -itd -p 80:8080 awsdeeplearningteam/mms_cpu
+    docker exec mms pip install -U -e /mxnet-model-server/.  >& /dev/null
+    docker exec mms mxnet-model-server start --mms-config /mxnet-model-server/docker/mms_app_cpu.conf >& /dev/null &
+    if [[ `echo $?` != 0 ]] ; then
+        echo "ERROR: Exiting as MMS startup failed"
+        remove_container_images
+        exit 1
+    fi
+    sleep 30
+}
+
+start_mms_gpu(){
+    nvidia-docker run --name mms -v /mxnet-model-server:/mxnet-model-server -itd -p 80:8080 awsdeeplearningteam/mms_gpu
+    nvidia-docker exec mms pip install -U -e /mxnet-model-server/. >& /dev/null
+    nvidia-docker exec mms pip uninstall --yes mxnet-cu90mkl  >& /dev/null
+    nvidia-docker exec mms pip uninstall --yes mxnet  >& /dev/null
+    nvidia-docker exec mms pip install mxnet-cu90mkl  >& /dev/null
+    nvidia-docker exec mms mxnet-model-server start --mms-config /mxnet-model-server/docker/mms_app_gpu.conf >& /dev/null &
+    if [[ `echo $?` != 0 ]] ; then
+        echo "ERROR: Exiting as MMS startup failed"
+        remove_container_images
+        exit 1
+    fi
+    sleep 90
+}
+
+start_jmeter(){
+    cd /mxnet-model-server/load-test
+    echo "$report"
+    ./run_load_test.sh -i 127.0.0.1 -c $workers -n $requests -f $report -p 80
+    if [[ `echo $?` != 0 ]] ; then
+        echo "ERROR: Exiting as jmeter startup failed"
+        remove_container_images
+        exit 1
+    fi
+    echo "done report generation"
+}
+
+remove_container_images(){
+    docker rm -f mms
+}
+
+run_test(){
+    export JMETER_HOME=$HOME/apache-jmeter-4.0
+    export PATH=$JMETER_HOME/bin:$PATH
+    for workers in 50 100
+    do
+        for requests in 25 50 100
+        do
+            echo "*************************************** Workers $workers and reqs $requests   ****************************************************"
+            if [ "$use_gpu" == "0" ]; then
+                start_mms_cpu
+            elif [ "$use_gpu" == "1" ]; then
+                start_mms_gpu
+            fi
+            start_jmeter
+            parse_csv_log
+            docker rm -f mms
+        done
+    done
+}
+
+
+main(){
+    install_docker
+    install_jmeter
```

Comment:
If we add "&&" to each of the running commands in the above methods, we should probably check for return values ("echo $?") after every step here.